### PR TITLE
Adds the Engineering Materiel Vacuum

### PR DESCRIPTION
**Pull request by BatElite**
[feat][qol]
## About the PR 
Adds the engineering materiel vacuum (or engivac), a tool for engineers that collects building materials and debris (anything reclaimable) as you walk over turfs and also from walls/windows you move into. 

-Takes any flavour of toolbox to put stuff into and take floortiles from.
-Fits on belt or back, to accomodate what's most useful to any user. Most non-engineering crew are more likely willing to have it on their belt slot (unutilised by some departments), while mechanics and engineers tend not to want to part with their toolbelts. Handheld use is also an option.
-Collecting building mats/debris as well as floortile placement can all be toggled.
-Placing floortiles is automatic as long as supply lasts, while replacing broken/burnt ones is automatic if you're holding a crowbar.

(NB I forgot to add these to engineering lockers and `also_replace_broken` can probably be a define instead of a var, but those are tiny enough tweaks I can put out tomorrow not to warrant making this a draft. The code is good to review as far as I'm concerned.)

-Since this sort of lets people carry a toolbox in their belt slot (though you need both hands to remove the engivac and then the toolbox, so it's clunky), I'm anticipating a concern that it gives people effectively a 7-slot fanny pack again. Personally I'd rather wait and see if it becomes a problem in practice but if need be, I can add a proc that rejects a toolbox with too many non-listed things in it and an emag interaction that disables that check.
-This PR uses `move_callback` and involves some copy-paste/ugly code to get the thing working. The whole situation surrounding adding/removing items to the move_callback list looks fucked up to me, but refactoring that is best left to its own PR. If it's not clear, the obj.dm and cable_coil.dm changes are just moving a proc on /obj/ to a more sensible place.

## Why's this needed? 
Engineers barely have a real edge over crew when it comes to building or station repairs (free carpenter as well as building mats in a place guaranteed to get broken into, woo), so they're in need of job-specific tools ~~including liberal RCD availability but I'm not prodding that debate right now~~. This one should make it easy to fix welderbomb and tile-prying staffie damage and to clean up all the crap that gets left lying around after a bombing.

## Changelog

```changelog
(u)BatElite
(*)Added the engineering materiel vacuum, a hand-or-back-or-belt item which allows them to collect loose debris and construction materials as well as place floor tiles by simply walking around with it!
```

### CODE REVIEW

**Comment by UrsulaMejor** [comment]
1) Where is this obtained?

2) No toolboxes on belt slots, please. I'd also prefer if the emag action made it do something more destructive or annoying than letting you use it is a fanny pack.

**Comment by BatElite** [comment]
It's in engineer's lockers but as I noted I forgot to actually put them in there at first. Also, non-list item limit is in

File: code/mob/living.dm [hunk 1]
```diff
@@ -1159,6 +1159,15 @@
 	. = ..()
 	if (isturf(oldloc) && isturf(loc) && move_laying)
 		var/list/equippedlist = src.equipped_list()
+
+		//This is terrible but "refactor all the move_laying garbage in my item PR" is worse
```
[the remark this answers is not in the document]
Comment (BatElite) [reply code/mob/living.dm]: Didn't know that existed before

**Comment by BatElite** [comment]
The floortile placing thing was something of an afterthought while making it, but IMO there's no reason to make it a separate tool when this one has most of the stuff in place to conveniently add it.

(also provided the checks work out this should be ready)

**Review by pali6 — APPROVED** [review]
If this proves to be too usable as just an additional storage thing on belt I think that could be solved at a later point by making it so inserting and removing the toolbox takes some amount of time imo.

**Pull request merged**